### ISSUE TITLE
labels should not be same as input_ids for casual language model

### ISSUE THREAD

**Issue by thomas010** [huggingface/transformers#24910]
### System Info

in the example for casual language mode pretraing (examples/pytorch/language-modeling/run_clm.py: line 490),
labels in 'result' should not be the same as input_ids, but maybe as follows
````
input_ids = result["input_ids"]
  result["input_ids"] = input_ids[:, :-1].contiguous()
  result["labels"]=input_ids[:, :-1].contiguous()
````

### Information

- [X] The official example scripts
- [ ] My own modified scripts

### Tasks

- [X] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

examples/pytorch/language-modeling/run_clm.py: line 490

### Expected behavior

update

**Comment by avivbrokman** [comment]
@thomas010 did you mean to provide the same code for `result['input_ids']` and `result['labels']`?

I don't know if this helps you, but in standard causal language modeling, the input sequence and labels _should_ be the same.  Suppose the sequence is <token_A>, <token_B>, <token_C>.  In step 0, we want to predict <token_A>.  In Step 1, we want to predict <token_B> from <token_A>.  In step 2, we want to predict <token_C> from <token_A>, <token_B>.  `transformers` knows this is what we want when we provide the same sequence as `input_ids` and `labels`.

**Comment by thomas010** [comment]
i got it, thank you

**Comment by glorialucheng** [comment]
> [@thomas010](https://github.com/thomas010) did you mean to provide the same code for `result['input_ids']` and `result['labels']`?
> 
> I don't know if this helps you, but in standard causal language modeling, the input sequence and labels _should_ be the same. Suppose the sequence is <token_A>, <token_B>, <token_C>. In step 0, we want to predict <token_A>. In Step 1, we want to predict <token_B> from <token_A>. In step 2, we want to predict <token_C> from <token_A>, <token_B>. `transformers` knows this is what we want when we provide the same sequence as `input_ids` and `labels`.

Can it be considered that the transformers framework will automatically build input_ids based on the sequence <token_A>, <token_B>, <token_C>: <start>, <token_A>, <token_B>, <token_C>, and build labels: <token_A>, <token_B>, <token_C>, <end>
